packages/edit-navigation/src/components/menu-locations-editor/index.js

```diff
@@ -1,3 +1,167 @@
+/**
+ * External dependencies
+ */
+import { map, flatMap, remove, find, pick, groupBy } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import {
+	Button,
+	Panel,
+	PanelBody,
+	SelectControl,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MenuSelectControl = ( { onChange, menusList, location } ) => {
+	const [ menuId, setMenuId ] = useState( false );
+
+	useEffect( () => {
+		setMenuId( location.menu );
+	}, [ location.menu ] );
+
+	return (
+		<SelectControl
+			options={ menusList }
+			value={ menuId }
+			onChange={ ( selectedMenuId ) => {
+				onChange( menuId, {
+					location: location.name,
+					menu: parseInt( selectedMenuId ),
+				} );
+				setMenuId( selectedMenuId );
+			} }
+		/>
+	);
+};
+
 export default function MenuLocationsEditor() {
-	return <>Menu locations editor</>;
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus() );
+
+	const [ menuLocations, setMenuLocations ] = useState( false );
+	const [ locationsData, setLocationsData ] = useState( {} );
+	const { saveMenu } = useDispatch( 'core' );
+
+	const kind = 'root';
+	const name = 'menuLocation';
+	const entities = useSelect( ( select ) =>
+		select( 'core' ).getEntitiesByKind( kind )
+	);
+	const entity = find( entities, { kind, name } );
+
+	const getMenuLocations = async () => {
+		const path = `${ entity.baseURL }`;
+		let apiLocations = await apiFetch( {
+			path,
+			method: 'GET',
+		} );
+		apiLocations = flatMap( apiLocations, ( value ) => value );
+		return apiLocations;
+	};
+
+	const setLatestLocations = async () => {
+		const latestLocations = await getMenuLocations();
+		setMenuLocations( latestLocations );
+	};
+
+	useEffect( () => {
+		setLatestLocations();
+	}, [] );
+
+	useEffect( () => {
+		if ( menus && menuLocations ) {
+			const locations = groupBy( menuLocations, 'menu' );
+			map( locations, ( location, menuId ) => {
+				locations[ menuId ] = map( location, 'name' );
+			} );
+			setLocationsData( locations );
+		}
+	}, [ menuLocations ] );
+
+	if ( ! menus || ! menuLocations ) {
```

Comment:
Do we really need to hide the whole UI if menus / menu locations are empty? In my testing it made the UI jump.

Reply:
Maybe not, but we don't have an empty state imagined so I'd rather this :)

Reply:
Okay, showing the spin when the page loads I get but I dont why it shows when you press save.

Reply:
+1. Even saying "You don't have any menus yet. Why don't you _create one_" would be better. CC @karmatosed